test/cppsuite/test_harness/workload/workload_validation.h

```diff
@@ -75,66 +81,80 @@ class workload_validation {
          * checked in check_reference.
          */
         for (auto const &it : deleted_collections) {
-            if (!verify_collection_state(session, it, false))
+            if (!verify_collection_file_state(session, it, false))
                 testutil_die(DEBUG_ERROR,
-                  "validate: collection %s present on disk while it has been tracked as deleted.",
+                  "Validation failed: collection %s present on disk while it has been tracked as "
+                  "deleted.",
                   database::build_collection_name(it).c_str());
         }
 
+        /*
+         * All collections in memory should match those created in the schema tracking table.
+         * Dropping is currently not supported.
+         */
+        std::sort(created_collections.begin(), created_collections.end());
+        auto on_disk_collection = created_collections.begin();
+        for (const auto &it : known_collection_ids) {
+            if (it != *on_disk_collection)
+                testutil_die(DEBUG_ERROR,
+                  "Validation failed: collection state mismatch expected "
+                  "collection id %lu but got %lu.",
+                  it, *on_disk_collection);
+            on_disk_collection++;
+        }
```

Comment:
Do you need to check that `on_disk_collection == created_collections.end()`?

Reply:
Using a size check instead. 8153016